sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/DataStreams.java

```diff
@@ -185,15 +186,19 @@ public DataStreamDecoder(Coder<T> coder, PrefetchableIterator<ByteString> inputS
      * Skips any remaining bytes in the current {@link ByteString} moving to the next {@link
      * ByteString} in the underlying {@link ByteString} {@link Iterator iterator} and decoding
      * elements till at the next boundary.
+     *
+     * @param weightEstimatorFn Pass in what is the function to be used for the weight.
      */
-    public List<T> decodeFromChunkBoundaryToChunkBoundary() {
+    public WeightedList<T> decodeFromChunkBoundaryToChunkBoundary(
+        Function<T, Long> weightEstimatorFn) {
       inbound.currentStream = inputByteStrings.next().newInput();
```

Comment:
Hmm that's true, although I don't think the overhead of keeping an extra counter adds too much.

Do you think weight per element isn't useful? I can think it would come handy in a few optimizations like avoid recomputing the sizes if an element was removed, but those are further optimizations / I didn't want to change more than necessary.

Reply:
Currently we're only keeping the total weight anyhow. (Keeping per-element weights seems to be a large expense, especially as the goal with these weights is to try to manage memory.)

Reply:
Understand. Makes sense, doing the change.